org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java

```diff
@@ -622,4 +634,152 @@ private static Block getBlockBodyForLambda(Expression bodyExpr, ITypeBinding ret
 		blockBody.statements().add(statementInBlockBody);
 		return blockBody;
 	}
+
+	public static boolean getCatchClauseToThrowsProposals(IInvocationContext context, ASTNode node, Collection<CUCorrectionProposal> resultingCollections) {
+		CatchClause catchClause = (CatchClause) ASTResolving.findAncestor(node, ASTNode.CATCH_CLAUSE);
+		if (catchClause == null) {
+			return false;
+		}
+
+		Statement statement = ASTResolving.findParentStatement(node);
+		if (statement != catchClause.getParent() && statement != catchClause.getBody()) {

```

Comment:
shouldn't you use equals(...) instead?

Reply:
@fbricon it is comparing 2 AST nodes. They should point to the same object instance in this case.